docs/architecture/ADR-005-QGB_posting_confirms_in_reserved_namespace.md

```diff
@@ -0,0 +1,96 @@
+This was done in the goal of not charging orchestrators increasing gas fees with every posted attestations.
+A simple benchmark showed that the gas usage multiplied 2 times from `~50 000` to `100 000` after submitting 16 attestation.
+Also, even if removing the checks was the most practical solution, it ended up opening new attack vectors on the QGB module state, such as flooding the network with incorrect attestations from users who are not even validators.
+Which would increase the burden on validators to handle all of that state.
+Furthermore, it put the responsibility on the relayer to cherry-pick the right confirms from the invalid ones.
+
+This led us to think that the direction we're taking makes more sense if it moves the confirms from the state to the block space.
+This way, we can remove all the complex logic handling the Valset Confirms and DataCommitment Confirms to the orchestrator side, and also block any opportunity for any malicious party to post data to state.
+
+In fact, we will be able to remove the `MsgValsetConfirm` defined in [here](https://github.com/celestiaorg/celestia-app/blob/a965914b8a467f0384b17d9a8a0bb1ac62f384db/proto/qgb/msgs.proto#L24-L49)
+And also, the `MsgDataCommitmentConfirm` defined in [here](
+https://github.com/celestiaorg/celestia-app/blob/a965914b8a467f0384b17d9a8a0bb1ac62f384db/proto/qgb/msgs.proto#L55-L76).
+Which were the way orchestrators were able to post confirms to the QGB module.
+Then, the only state that the QGB module will keep is the one that is created in [EndBlocker](https://github.com/celestiaorg/celestia-app/blob/a965914b8a467f0384b17d9a8a0bb1ac62f384db/x/qgb/abci.go#L12-L16).
+Which are `Attestations`, i.e. `Valset`s and `DataCommitmentRequest`s.
+We will need to investigate the state's growth if we keep in it only these latter.
+
+In addition to this, we will need to reserve a certain namespace ID to only be reserved for Confirms.
+This might also require a discussion on how we want to put the data in the block, and investigate the ability to batch confirms.
+
+When it comes to slashing, we can add the `dataRoot` of the blocks to state during `ProcessProposal`,  `FinalizeCommit`, or in some other way to be defined. Then, we will have a way to slash orchestrators after a certain period of time if they didn't post any confirms. The exact details of this will be left for another ADR.
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Positive
+
+- Reduce significantly the gas fees paid by orchestrators.
+- Reduce significantly the use of Celestia state.
+- Keep the whole QGB confirms history via using block data, instead of being forced to delete it after the unbonding period not to end up with a super gigantic state. This would allow any QGB contract to have the whole history on any EVM chain.
```

Comment:
[nit]
```suggestion
- Keep the whole QGB confirms history via using block data, instead of being forced to delete it after the unbonding period to not end up with a super gigantic state. This would allow any QGB contract to have the whole history on any EVM chain.
```